ls1intum/Artemis
# `General`: Shorten the display of the result score

### Checklist
#### General


#### Client
#### Changes affecting Programming Exercises

### Motivation and Context


The UI should not be overloaded with unnecessary information that might take up too much space. Therefore the result string is now even shorter inn certain places

### Description

The course row, exercise header, result timeline in the exercise view, the code editor, ... now use the shorter format that only shows the achieved score. When clicking on the result, the achieved points as well as the amount of passed tests is shown.
Additionally the table showing all results in the student view of a programming exercise looks nicer.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Student
- 1 Programming Exercise

1. Log in to Artemis
2. Participate in the Programming exercise and look at every visible result string
3. Check that it displays the correct values and that the UI is not too cluttered.
4. Check the detail view of the results by clicking on it.

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage


| Class/File | Branch | Line |
|------------|-------:|-----:|
| result.service.ts | 86% | 76% |


### Screenshots


![Bildschirmfoto 2022-10-13 um 17 56 20](https://user-images.githubusercontent.com/38322605/195646124-dd36aea4-63c1-469b-91ec-0a860213ee76.png)
![Bildschirmfoto 2022-10-11 um 13 27 42](https://user-images.githubusercontent.com/38322605/195078225-d7611b62-d773-4836-833f-24ccc2d63fbe.png)
